### ISSUE TITLE
Missing support for `migrate --force`

### ISSUE THREAD

**Issue by mrsimonemms** [riggerthegeek/east-mysql#1]
From @okv: `markExecuted` method should allow mark of already marked transaction to support `migrate --force`. I've just added test for that to the east repository.

**Comment by mrsimonemms** [comment]
@okv - really not sure what you mean by this.  The `migrate --force` works quite cheerfully and it does mark it in the migration table.  Also, I looked in the other adapters and they don't specify that a migration was forced.

I'll mark it as closed, but if I've missed the point entirely, let me know and I'll sort it out.

**Comment by okv** [comment]
try to execute same migration twice (with --force flag at second time). Currently with mysql adapter you will get an error `Error: ER_DUP_ENTRY: Duplicate entry '1_someMigration' for key 'PRIMARY'. You can run tests from east repository with mysql adapter(i've added section "Creating and testing own adapter" on east page). They checks that case too.

**Comment by mrsimonemms** [comment]
Right, I think I see what's going on here.  The error `Error: ER_DUP_ENTRY: Duplicate entry '1_someMigration' for key 'PRIMARY` is a MySQL error because the `1_someMigration` file is trying to insert a new line with an already existing primary key.  As the `east-mysql` package (as all the adapters) are only interested in running the commands they are given, in order to run this you'll need to change the test to not set the primary key.

Incidentally, I've run the `east` tests on `east-mysql` and I'm getting that error.  However, I'm struggling to find the actual SQL that's being executed.  Part of the reason for this seems to be that the unit tests are not stateless and need to be run in a specific order - when I add `.only` to the unit tests that's failing, it fails for an entirely different reason `Error: Cannot find module '/Users/SimonEmms/Documents/Development/Personal/east/test/migrations/undefined.js'` which makes debugging very difficult.  If I was designing the tests for the `--force` functionality, I would probably set the test up so that it runs the script once then reruns it with `--force`.  That way, the tests can be run out of order or alone.

If you are able to tell me the SQL that's failing (or how to find it), I can come up with a proposal for a unit test that won't fail.

For the record, I know that `--force` works as I've tried it with the following script which works perfectly cheerfully:

```
INSERT INTO users SET email='test@test.com', firstname='Test', lastname='Testington'
```

**Comment by okv** [comment]
east by itself doesn't execute any db commands (sql or mongodb queries), he relies on adapters. So in case of --force flag `markExecuted` will be called twice with same `name` and east intents that adapter just do nothing when you call mark executed for migration that already marked as executed
e.g. in postgres adapter duplicate key error will be silently omitted by error code 23505

``` js
this.db.query('insert into _migrations values($1)', [name], function(err) {
        callback(err && err.code !== '23505' ? err : null);
});
```

**Comment by okv** [comment]
I've changed east logic now `markExecuted` will not be called for already marked migration when force flag is set. So since east 0.2.5 it works like you expected. Could you publish version with fixed mysql dependency to npm?

**Comment by mrsimonemms** [comment]
Ok

Just published v0.2.0 with the dependency fix (apologies, I thought I'd already done this)
